prometheus/src/main/scala/filodb/prometheus/ast/Vectors.scala

```diff
@@ -139,12 +138,57 @@ sealed trait Vector extends Expression {
   val regexColumnName: String = "::(?=[^::]+$)" //regex pattern to extract ::columnName at the end
 
   // Convert metricName{labels} -> {labels, __name__="metricName"} so it's uniform
-  lazy val mergeNameToLabels: Seq[LabelMatch] = {
+  // Note: see makeMergeNameToLabels before changing the laziness.
+  lazy val mergeNameToLabels: Seq[LabelMatch] = makeMergeNameToLabels()
+
+  // These constraints are also applied within makeMergeNameToLabels, but they are
+  //   *immediately* applied here (i.e. not lazily as with mergeNameToLabels).
+  applyPromqlConstraints()
+
+  /**
+   * Apply PromQL-scope constraints to the vector selector:
+   *   (1) only one __name__ specifier (implicit or otherwise) exists.
+   *   (2) at least one label matcher exists.
```

Comment:
This should technically be "at least one label matcher does not match the empty string," but there isn't a clean way to enforce this with this setup-- to use a `ColumnFilter`, `mergeNameToLabels` needs to be evaluated.

Reply:
(this validation exists to pass a `parseError("{}")` test)